packages/react-devtools-shared/src/utils.js

```diff
@@ -582,66 +583,73 @@ export function getDataType(data: Object): DataType {
+        return 'bigint';
+      case 'boolean':
+        return 'boolean';
+      case 'function':
+        return 'function';
+      case 'number':
+        if (Number.isNaN(data)) {
+          return 'nan';
+        } else if (!Number.isFinite(data)) {
+          return 'infinity';
         } else {
-          return iterator === data ? 'opaque_iterator' : 'iterator';
+          return 'number';
         }
-      } else if (data.constructor && data.constructor.name === 'RegExp') {
-        return 'regexp';
-      } else {
-        const toStringValue = Object.prototype.toString.call(data);
-        if (toStringValue === '[object Date]') {
-          return 'date';
-        } else if (toStringValue === '[object HTMLAllCollection]') {
+      case 'object':
+        if (Array.isArray(data)) {
+          return 'array';
+        } else if (ArrayBuffer.isView(data)) {
+          return hasOwnProperty.call(data.constructor, 'BYTES_PER_ELEMENT')
+            ? 'typed_array'
+            : 'data_view';
+        } else if (
+          data.constructor &&
+          data.constructor.name === 'ArrayBuffer'
+        ) {
+          // HACK This ArrayBuffer check is gross; is there a better way?
+          // We could try to create a new DataView with the value.
+          // If it doesn't error, we know it's an ArrayBuffer,
+          // but this seems kind of awkward and expensive.
+          return 'array_buffer';
+        } else if (typeof data[Symbol.iterator] === 'function') {
+          const iterator = data[Symbol.iterator]();
+          if (!iterator) {
+            // Proxies might break assumptoins about iterators.
+            // See github.com/facebook/react/issues/21654
+          } else {
+            return iterator === data ? 'opaque_iterator' : 'iterator';
+          }
+        } else if (data.constructor && data.constructor.name === 'RegExp') {
+          return 'regexp';
+        } else {
+          const toStringValue = Object.prototype.toString.call(data);
+          if (toStringValue === '[object Date]') {
+            return 'date';
+          } else if (toStringValue === '[object HTMLAllCollection]') {
+            return 'html_all_collection';
+          }
+        }
+        return 'object';
+      case 'string':
+        return 'string';
+      case 'symbol':
+        return 'symbol';
+      case 'undefined':
+        if (
+          Object.prototype.toString.call(data) === '[object HTMLAllCollection]'
+        ) {
           return 'html_all_collection';
         }
-      }
-      return 'object';
-    case 'string':
-      return 'string';
-    case 'symbol':
-      return 'symbol';
-    case 'undefined':
-      if (
-        Object.prototype.toString.call(data) === '[object HTMLAllCollection]'
-      ) {
-        return 'html_all_collection';
-      }
-      return 'undefined';
-    default:
-      return 'unknown';
+        return 'undefined';
+      default:
+        return 'unknown';
+    }
+  } catch (ignore) {
```

Comment:
It's not great to wrap the whole thing in try/catch because that will miss actual bugs in DevTools code. Can you wrap just the parts you expect to throw?